test/formatter.test.js

```diff
@@ -130,6 +134,30 @@ test('GH-845: sync formatter should blow up', function(t) {
     );
 });
 
+test('sync formatter should handle expected errors gracefully', function(t) {
+    var hasUncaughtException = false;
+    SERVER.on('uncaughtException', function(req, res, route, err) {
+        hasUncaughtException = true;
```

Comment:
nit: How about adding an `assert.throw('Should not reach')` instead of the boolean logic which is more fragile to execution order.